java/code/src/com/suse/oval/parser/OvalParser.java

```diff
@@ -0,0 +1,724 @@
+        ovalRootType.setDefinitions(allDefinitions);
+        ovalRootType.setObjects(ovalResources.getObjects());
+        ovalRootType.setStates(ovalResources.getStates());
+        ovalRootType.setTests(ovalResources.getTests());
+
+        return ovalRootType;
+    }
+
+    /**
+     * Parses the given OVAL file in bulks. For every bulk parsed, it calls {@link OVALDefinitionsBulkHandler#handle}.
+     *
+     * @param ovalFile an XML file containing OVAL definitions to be parsed.
+     * @param bulkHandler an operation to applied on every bulk of parsed OVAL definitions.
+     * */
+    public void parseDefinitionsInBulk(File ovalFile, OVALDefinitionsBulkHandler bulkHandler) {
+        XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+        // Disable external entities processing as a protection measure against XXE.
+        xmlInputFactory.setProperty("javax.xml.stream.isSupportingExternalEntities", false);
+
+        try {
+            XMLEventReader2 reader =
+                    (XMLEventReader2) xmlInputFactory.createXMLEventReader(new FileInputStream(ovalFile));
+            // Disable external entities processing as a protection measure against XXE.
+            xmlInputFactory.setProperty("javax.xml.stream.isSupportingExternalEntities", false);
+
+            List<DefinitionType> definitions = new ArrayList<>();
+
+            while (reader.hasNext()) {
+                XMLEvent2 nextEvent = (XMLEvent2) reader.nextEvent();
+
+                if (nextEvent.isStartElement() &&
+                        nextEvent.asStartElement().getName().getLocalPart().equals("definition")) {
+                    DefinitionType definitionType = parseDefinitionType(nextEvent.asStartElement(), reader);
+                    definitions.add(definitionType);
+
+                    if (definitions.size() == DEFINITIONS_BULK_SIZE) {
+                        bulkHandler.handle(definitions);
+                        definitions = new ArrayList<>();
+                    }
+                }
+
+                if (nextEvent.isEndElement() &&
+                        nextEvent.asEndElement().getName().getLocalPart().equals("definitions")) {
+                    if (!definitions.isEmpty()) {
+                        bulkHandler.handle(definitions);
+                    }
+                    break;
+                }
+            }
+        }
+        catch (XMLStreamException | FileNotFoundException e) {
+            throw new OvalParserException("Failed to parse OVAL definitions from OVAL file at: " +
+                    ovalFile.getAbsolutePath(), e);
+        }
+    }
+    /**
+     * Utility method to parse all OVAL definitions at once instead of in bulks. To be used in testing.
+     *
+     * @param ovalFile an XML file containing OVAL definitions to be parsed.
+     * @return all OVAL definitions in {@code ovalFile}
+     * */
+    public List<DefinitionType> parseAllDefinitions(File ovalFile) {
```

Comment:
This is a dependency for the parser function that is used only for testing. We want people to use the bulk parser, and not the one that loads everything to memory.